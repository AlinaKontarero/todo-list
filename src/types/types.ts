export type ITask = {
  content: string
  isHighPriority: boolean
  isCompleted: boolean
}

export type Order = 'asc' | 'desc';

export type OrderBy = 'name' | 'priority'