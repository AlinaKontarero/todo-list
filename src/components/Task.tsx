import * as React from 'react'
import { Tooltip, IconButton, Checkbox } from '@material-ui/core'
import DeleteIcon from '@material-ui/icons/Delete';
import { Zoom } from '@material-ui/core';
import ArrowUpwardIcon from '@material-ui/icons/ArrowUpward';
import ArrowDownwardIcon from '@material-ui/icons/ArrowDownward';
import { ITask } from '../types/types';


interface Props {
  task: ITask
  onDelete: (content: string) => void 
  handleComplete: (content: string) => void
  handlePriority: (content: string) => void
}

class Task extends React.Component<Props, never> { 
  render() {
    const { task, handleComplete, onDelete, handlePriority } = this.props

    return (
        <div className='columns is-variable is-2 is-vcentered'>
          <div className='column is-narrow'>
            <Checkbox 
              value={task.content}
              checked={task.isCompleted}
              onChange={(e: React.ChangeEvent<HTMLInputElement>) => handleComplete(e.target.value)}
            />
          </div>
          <div className='column'>
            {task.content}
          </div>
          <div className='column is-narrow'>
          <Tooltip title='Priority' TransitionComponent={Zoom} arrow={true} placement={'top'}>
            <IconButton 
              onClick={() => handlePriority(task.content)} 
              color='inherit'
              >
              {task.isHighPriority ? <ArrowUpwardIcon /> : <ArrowDownwardIcon /> }
            </IconButton>
          </Tooltip>
          </div>
          <div className='column is-narrow is-pulled-right'>
          <Tooltip title='Close' TransitionComponent={Zoom} arrow={true} placement={'top'}>
            <IconButton 
              onClick={() => onDelete(task.content)} 
              color='inherit'
              >
              <DeleteIcon />
            </IconButton>
          </Tooltip>
          </div>
        </div>
    )
  }
}

export default Task