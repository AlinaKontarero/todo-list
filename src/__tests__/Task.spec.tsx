import * as React from 'react'
import {shallow} from 'enzyme'
import { ITask } from '../types/types'
import Task from '../components/Task'

describe('Task', () => {
  const mockTask: ITask = {
      content: 'Test 1',
      isCompleted: true,
      isHighPriority: true};
   
  const taskComponent = <Task
    task={mockTask}
    onDelete={() => console.log('handleDelete')}
    handleComplete={() => console.log('handleComplete')}
    handlePriority={() => console.log('handlePriority')}
  />
  it('should render correctly', () => {
    const wrapper = shallow(taskComponent);
  })
  it("should render initial layout", () => {
    // when
    const component = shallow(taskComponent);
    // then
    expect(component.getElements()).toMatchSnapshot();
  });
})