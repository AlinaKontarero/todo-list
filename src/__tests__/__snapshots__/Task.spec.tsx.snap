// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task should render initial layout 1`] = `
Array [
  <div
    className="column is-full is-task"
  >
    <div
      className="columns is-variable is-2 is-vcentered is-gapless"
    >
      <div
        className="column is-narrow"
      >
        <WithStyles(ForwardRef(Checkbox))
          checked={true}
          onChange={[Function]}
          value="Test 1"
        />
      </div>
      <div
        className="column"
      >
        Test 1
      </div>
      <div
        className="column is-narrow"
      >
        <WithStyles(ForwardRef(Tooltip))
          TransitionComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "children": [Function],
                "disableStrictModeCompat": [Function],
                "in": [Function],
                "onEnter": [Function],
                "onEntered": [Function],
                "onEntering": [Function],
                "onExit": [Function],
                "onExited": [Function],
                "onExiting": [Function],
                "style": [Function],
                "timeout": [Function],
              },
              "render": [Function],
            }
          }
          arrow={true}
          placement="top"
          title="Priority"
        >
          <WithStyles(ForwardRef(IconButton))
            color="inherit"
            onClick={[Function]}
          >
            <Memo />
          </WithStyles(ForwardRef(IconButton))>
        </WithStyles(ForwardRef(Tooltip))>
      </div>
      <div
        className="column is-narrow is-pulled-right"
      >
        <WithStyles(ForwardRef(Tooltip))
          TransitionComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "propTypes": Object {
                "children": [Function],
                "disableStrictModeCompat": [Function],
                "in": [Function],
                "onEnter": [Function],
                "onEntered": [Function],
                "onEntering": [Function],
                "onExit": [Function],
                "onExited": [Function],
                "onExiting": [Function],
                "style": [Function],
                "timeout": [Function],
              },
              "render": [Function],
            }
          }
          arrow={true}
          placement="top"
          title="Close"
        >
          <WithStyles(ForwardRef(IconButton))
            color="inherit"
            onClick={[Function]}
          >
            <Memo />
          </WithStyles(ForwardRef(IconButton))>
        </WithStyles(ForwardRef(Tooltip))>
      </div>
    </div>
  </div>,
]
`;
